ddtrace/internal/sampling.py

```diff
@@ -208,44 +208,60 @@ def apply_span_sampling_tags(self, span):
+            except FileNotFoundError:
+                raise ValueError("Unable to parse DD_SPAN_SAMPLING_RULES_FILE=%r" % file_json_rules)
+    # No rules specified
     else:
-        sampling_rules = []
-        try:
-            json_rules = json.loads(json_rules_raw)  # type: List[SpanSamplingRules]
-            if not isinstance(json_rules, list):
-                raise TypeError("DD_SPAN_SAMPLING_RULES is not list, got %r" % json_rules)
-        except JSONDecodeError:
-            raise ValueError("Unable to parse DD_SPAN_SAMPLING_RULES=%r" % json_rules_raw)
-        for rule in json_rules:
-            if not isinstance(rule, dict):
-                raise TypeError("rule specified via DD_SPAN_SAMPLING_RULES is not a dictionary:%r" % rule)
-            # If sample_rate not specified default to 100%
-            sample_rate = float(rule.get("sample_rate", 1.0))
-            service = rule.get("service")
-            name = rule.get("name")
-            # If max_per_second not specified default to no limit
-            max_per_second = int(rule.get("max_per_second", -1))
-            if service is None and name is None:
-                raise ValueError(
-                    "Neither service or name specified for single span sampling rule:%r,"
-                    "at least one of these must be specified" % rule
-                )
-            if service:
-                _check_unsupported_pattern(service)
-            if name:
-                _check_unsupported_pattern(name)
+        return []
 
-            try:
-                sampling_rule = SpanSamplingRule(
-                    sample_rate=sample_rate, service=service, name=name, max_per_second=max_per_second
-                )
-            except Exception as e:
-                raise ValueError("Error creating single span sampling rule {}: {}".format(json.dumps(rule), e))
-            sampling_rules.append(sampling_rule)
-        return sampling_rules
+    try:
+        json_rules = json.loads(raw_json_rules)
+        if not isinstance(json_rules, list):
+            raise TypeError("DD_SPAN_SAMPLING_RULES is not list, got %r" % json_rules)
+    except JSONDecodeError:
+        raise ValueError("Unable to parse DD_SPAN_SAMPLING_RULES=%r" % env_json_rules)
+
+    sampling_rules = []
+    for rule in json_rules:
+        if not isinstance(rule, dict):
+            raise TypeError("rule specified via DD_SPAN_SAMPLING_RULES is not a dictionary:%r" % rule)
+        # If sample_rate not specified default to 100%
+        sample_rate = float(rule.get("sample_rate", 1.0))
+        service = rule.get("service")
+        name = rule.get("name")
+        # If max_per_second not specified default to no limit
+        max_per_second = int(rule.get("max_per_second", -1))
+        if service is None and name is None:
+            raise ValueError("Neither service or name specified for single span sampling rule:%r" % rule)
+        if service:
+            _check_unsupported_pattern(service)
```

Comment:
I didn't catch this before, but we should validate that service and name are actually strings, we have our typing say they are, but they could be anything.

Reply:
reason why this would be nice is because if someone gives you a number then `for char in 10:` will fail and they'll get a weird `TypeError: 'int' object is not iterable`